moveit_planners/pilz_industrial_motion_planner/src/joint_limits_aggregator.cpp

```diff
@@ -39,22 +39,30 @@
 #include <moveit/robot_model_loader/robot_model_loader.h>
 
 #include <vector>
+namespace
+{
+static const rclcpp::Logger LOGGER =
+    rclcpp::get_logger("moveit.pilz_industrial_motion_planner.joint_limits_aggregator");
+}
 
 pilz_industrial_motion_planner::JointLimitsContainer
 pilz_industrial_motion_planner::JointLimitsAggregator::getAggregatedLimits(
-    const ros::NodeHandle& nh, const std::vector<const moveit::core::JointModel*>& joint_models)
+    const rclcpp::Node::SharedPtr& node, const std::string& param_namespace,
+    const std::vector<const moveit::core::JointModel*>& joint_models)
 {
   JointLimitsContainer container;
+  // TODO(henningkayser): use param_namespace
 
-  ROS_INFO_STREAM("Reading limits from namespace " << nh.getNamespace());
+  RCLCPP_INFO_STREAM(LOGGER, "Reading limits from namespace " << node->get_namespace());
 
   // Iterate over all joint models and generate the map
   for (auto joint_model : joint_models)
   {
     JointLimit joint_limit;
 
-    // If there is something defined for the joint on the parameter server
-    if (joint_limits_interface::getJointLimits(joint_model->getName(), nh, joint_limit))
+    // If there is something defined for the joint in the node parameters
+    if (::joint_limits_interface::declare_parameters(joint_model->getName(), node) &&
```

Comment:
Is it necessary to declare the parameters here (again?). I got errors due to already declared parameters which caused the PTP planner failing.